test/sharness/lib/test-lib.sh

```diff
@@ -312,3 +312,33 @@ test_str_contains() {
 	shift
 	echo "$@" | grep "$find" >/dev/null
 }
+
+test_links_in_trickle_dag() {
+	local LINKS="$(ipfs object links $1)"
+	local i="0"
+	echo "$LINKS" | while IFS= read -r line; do
+	    if [ $i -gt 0 ]; then
+            test_link_in_trickle_dag $(expr $i - 1) $line || return 1
+        fi
+		i=$(expr $i + 1)
+	done
+}
+
+test_link_in_trickle_dag() {
```

Comment:
(@whyrusleeping part of a reason to have it here is to allow other implementations of it that can be tested equally thoroughly)